wallet-new/integration/TransactionSpecs.hs

```diff
@@ -103,251 +109,232 @@ transactionSpecs wRef wc = do
+                                      encSecretKey
+        derivedPK = encToPublic derivedEncSK
+        addressCreatedFromThisPK = Core.checkPubKeyAddress derivedPK anAddress
+
+    addressCreatedFromThisPK `shouldBe` True
+
+    pure (anAddress, encToSecret derivedEncSK, derivedPK)
+
+
+makeAnotherAddress :: HasCallStack => EncryptedSecretKey -> Account -> IO Address
+makeAnotherAddress encSecretKey anAccount = do
+    addrIndex <- randomAddressIndex
+    let forBootstrapEra = Core.IsBootstrapEraAddr True
+        passCheck = ShouldCheckPassphrase True
+        Just (anAddress, derivedEncSecretKey) =
+            Core.deriveLvl2KeyPair forBootstrapEra
+                                   passCheck
+                                   emptyPassphrase
+                                   encSecretKey
+                                   (accIndex anAccount)
+                                   addrIndex
+        derivedPublicKey = encToPublic derivedEncSecretKey
+        addressCreatedFromThisPK = Core.checkPubKeyAddress derivedPublicKey anAddress
+
+    addressCreatedFromThisPK `shouldBe` True
+
+    pure anAddress
+
+
+storeAddressInWalletAccount :: HasCallStack => WalletClient IO -> Wallet -> Account -> Address -> IO ()
+storeAddressInWalletAccount wc wallet anAccount anAddress = do
+    -- Store this HD-address in the wallet's account.
+    let anAddressAsBase58 = Core.addrToBase58Text anAddress
+    void $ shouldReturnRight $ postStoreAddress wc
+                                                (walId wallet)
+                                                (accIndex anAccount)
+                                                anAddressAsBase58
+
+makeExternalWalletBasedOn :: HasCallStack => WalletClient IO -> PublicKey -> IO (Wallet, Account)
+makeExternalWalletBasedOn wc publicKey = do
+    newExtWallet <- randomExternalWalletWithPublicKey CreateWallet publicKey
+    extWallet <- createExternalWalletCheck wc newExtWallet
+    defaultAccount <- firstAccountInExtWallet wc extWallet
+    pure (extWallet, defaultAccount)
+
+makeWalletRootKeys :: IO (EncryptedSecretKey, PublicKey)
+makeWalletRootKeys = do
+    rootSK <- randomSK
+    let encRootSK = noPassEncrypt rootSK
+        rootPK    = encToPublic encRootSK
+    pure (encRootSK, rootPK)
+
+randomSK :: IO SecretKey
+randomSK = generate arbitrary
+
+randomAddressIndex :: IO Word32
+randomAddressIndex = generate arbitrary
+
+
+makePaymentFromExternalWallet :: HasCallStack
+                              => WalletClient IO
+                              -> Word64
+                              -> Wallet
+                              -> Account
+                              -> Wallet
+                              -> Address
+                              -> Address
+                              -> Address
+                              -> SecretKey
+                              -> PublicKey
+                              -> PublicKey
+                              -> IO ()
```

Comment:
For type signatures that overflow, indent the first line to provide more robust vertical alignment. eg:

```haskell
foobar 
    :: A 
    -> B 
    -> C
```